### ISSUE TITLE
bug: autocomplete doesn't work or works incorrectly with iTerm floatable window

### ISSUE THREAD

**Issue by danulqua** [withfig/fig#2525]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

macOS Ventura 13.3.11 (a)

### Expected behaviour

Open iTerm via dedicated hotkey, start typing in terminal, for example, `cd`, and see Fig's autocompletion popup to show up.

### Actual behaviour

Open iTerm via dedicated hotkey, start typing `cd` in terminal, and Fig's autocompletion not showing.
Current workaround - open default macOS Terminal app, start typing, and then you can go back to iTerm and Fig will continue working.

### Steps to reproduce

1. Go to iTerm settings
2. Go to Profiles tab
3. Inside Profiles tab go to Window tab and select Style: Full-Width Top of Screen
4. Inside Profiles tab go to Keys tab, turn on "A hotkey opens a dedicated window with this profile" and setup a hotkey under "Configure Hotkey Window"
5. Now, when you open iTerm via dedicated hotkey, Fig will stop working eventually
6. (workaround) When it stops working, open default macOS Terminal app, type something so Fig's autocompletion shows up, and then in iTerm it will work too

### Environment

```yaml
fig-details:
  - 2.16.0
hardware-info:
  - model:
  - model-id:
  - chip-id: Apple M1
  - cores: 8
  - mem: 16.00 GB
os-info:
  - macOS 13.3.1 (22E772610a)
environment:
  - shell: /opt/homebrew/Cellar/zsh/5.9/bin/zsh
  - terminal: iterm
  - cwd: /Users/danil0110/dev/antago/fuzzy-events
  - exe-path: /Users/danil0110/.fig/bin/fig
  - install-method: unknown
  - env-vars:
    - FIGTERM_SESSION_ID: 87b766d9-5ac3-47b2-be8d-fa0786e8c27b
    - FIG_SET_PARENT_CHECK: 1
    - FIG_TERM: 2.16.0
    - PATH: /opt/homebrew/bin:/opt/homebrew/sbin:/Users/danil0110/.oh-my-zsh/custom/plugins/git-open:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/danil0110/.fig/bin:/Users/danil0110/.local/bin
    - SHELL: /opt/homebrew/bin/zsh
    - TERM: xterm-256color
    - __CFBundleIdentifier: com.googlecode.iterm2
    - FIG_PID: 71465
    - FIG_SET_PARENT: 87b766d9-5ac3-47b2-be8d-fa0786e8c27b
```

**Comment by xshapira** [comment]
Having the same issue.

**Comment by AkinJimoh** [comment]
I'm experiencing a similar problem where auto-complete has completely stopped working, even though it was functioning fine until around 2 days ago.

**Comment by ginixsan** [comment]
same here..

**Comment by DJTB** [comment]
Same issue with floatable window. 
Was working fine until recently.

**Comment by pfurini** [comment]
Same here, iterm latest beta ver with macos intel 12.6.5. In addition I'm experiencing (intermittently) this in the last few days: https://github.com/withfig/fig/issues/2544
The popup sometimes appears in the lower left corner of the iterm window instead of near the cursor.

**Comment by danulqua** [comment]
any update?

**Comment by smacintyre** [comment]
Yup, experiencing the same here. Except it only ever works the very first time. When it does show up, the dropdown appears outside of the window:
<img width="1871" alt="image" src="https://github.com/withfig/fig/assets/361689/101c7b1b-2d09-4db9-aa44-18d123e49f41">

**Comment by macseem** [comment]
same issue, workaround with opened terminal works though, thanks for the tip @danulqua
